sys/trickle/trickle.c

```diff
@@ -30,63 +27,61 @@ void trickle_callback(trickle_t *trickle)
     if ((trickle->c < trickle->k) || (trickle->k == 0)) {
         (*trickle->callback.func)(trickle->callback.args);
     }
+
+    trickle_interval(trickle);
 }
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t max_interval;
+    uint32_t old_interval = trickle->I;
```

Comment:
still don't see the benefit, regarding your references:

[1] is equal to `diff = trickle->I - tickle->t`
[3] is equal to `trickle->t = random_uint32_range(trickle->I/2, trickle->I);`